pkg/build/generator/generator.go

```diff
@@ -211,6 +211,9 @@ func (g *BuildGenerator) checkLastVersion(bc *buildapi.BuildConfig, lastVersion
 // updateImageTriggers sets the LastTriggeredImageID on all the ImageChangeTriggers on the BuildConfig and
 // updates the From reference of the strategy if the strategy uses an ImageStream or ImageStreamTag reference
 func (g *BuildGenerator) updateImageTriggers(ctx kapi.Context, bc *buildapi.BuildConfig, from, triggeredBy *kapi.ObjectReference) error {
+	if triggeredBy == nil {
```

Comment:
yes, we should catch it in validation, but we don't want to panic on existing data. belt and suspenders